packages/components/src/color-palette/index.tsx

```diff
@@ -196,10 +207,10 @@ export default function ColorPalette( {
 	value,
 	__experimentalHasMultipleOrigins = false,
 	__experimentalIsRenderedInSidebar = false,
-} ) {
+}: ColorPaletteProps ) {
 	const clearColor = useCallback( () => onChange( undefined ), [ onChange ] );
 	const showMultiplePalettes =
-		__experimentalHasMultipleOrigins && colors?.length;
+		__experimentalHasMultipleOrigins && !! colors?.length;
```

Comment:
The type of `colors?.length` is `number`, so this ensures both `__experimentalHasMultipleOrigins` and `!! colors?.length` have the type of `boolean`